# Group Name - Jupyter FC
Team Memebers:Brian Mooney, Alex o'Brien, Sankalp Raj Srivastava


# Problem Statement
It’s your turn to be the manager/ scout/ recruiter . . .Now you need to put your analytics skills to the test, applying them to the problem of creating a ‘Dream Team’ . Your goal is to find a team of 11 players which could rival that of any primer league squad. 
Rules: 
I.You need to follow a 4-4-2 formation plus a goal keeper. 
II.You have a total market value of €80 Million Euro

In [4]:
#Uploading the CSV file on Google Collab
from google.colab import files
uploaded = files.upload()

Saving PlayerData.csv to PlayerData (2).csv


In [6]:
#Creating DataFrame for the CSV
dataframe = pd.read_csv('PlayerData (2).csv')

In [7]:
dataframe.head()  

Player                   Team  ... Penalties taken Penalty conversion, %
0   F. Uscanga               Tlaxcala  ...               4                  50.0
1     V. Henry                  Rodez  ...               0                   0.0
2     A. Turan             Ankaragücü  ...               0                   0.0
3   P. Bourdin      Beerschot-Wilrijk  ...               0                   0.0
4  C. Villalba  Talleres Córdoba Res.  ...               0                   0.0

[5 rows x 113 columns]

In [8]:
#Checking null values
dataframe.isnull().sum()

Player                              0
Team                                3
Position                            0
1st Position                        0
Age                                 0
                                 ... 
Direct free kicks per 90          310
Direct free kicks on target, %    310
Corners per 90                     20
Penalties taken                     0
Penalty conversion, %               0
Length: 113, dtype: int64

In [11]:
#DataFrame for GoalKeepers
GK_df = dataframe[(dataframe['1st Position'] ==  "GK")]

In [12]:
GK_df.head()

Player                  Team  ... Penalties taken Penalty conversion, %
6    G. Espinoza  Cimarrones de Sonora  ...               0                   0.0
19     P. Patron               Le Mans  ...               0                   0.0
90        M. Lis          Wisła Kraków  ...               0                   0.0
124  B. Siegrist         Dundee United  ...               0                   0.0
127  E. Volynets       Kolos Kovalivka  ...               0                   0.0

[5 rows x 113 columns]

In [25]:
#Ranking GoalKeepers based on Conceded goals per 90
GK_df["Rank"] = GK_df["Conceded goals per 90"].rank()
GK_df.head()

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """Entry point for launching an IPython kernel.


Player                  Team  ... Penalty conversion, %   Rank
6    G. Espinoza  Cimarrones de Sonora  ...                   0.0  104.0
19     P. Patron               Le Mans  ...                   0.0   71.5
90        M. Lis          Wisła Kraków  ...                   0.0  159.5
124  B. Siegrist         Dundee United  ...                   0.0  119.0
127  E. Volynets       Kolos Kovalivka  ...                   0.0   79.0

[5 rows x 114 columns]

In [29]:
#Sorting values against the features selected for Goalkeeper.
GK_df = GK_df.sort_values(["Shots against per 90", "Clean sheets","Save rate, %","Prevented goals per 90","Conceded goals per 90"], ascending = (True, False,True,False,True))

In [30]:
GK_df.head(20)

Player             Team  ... Penalty conversion, %   Rank
2046  Luís Maximiano      Sporting CP  ...                   0.0   10.5
4729      Z. Steffen  Manchester City  ...                   0.0    2.0
675       R. Majecki           Monaco  ...                   0.0    1.0
1143        A. Rossi     Boca Juniors  ...                   0.0    4.0
512        V. Barkas           Celtic  ...                   0.0   37.0
1497       A. Desmas         Clermont  ...                   0.0    5.5
5124      B. Lecomte           Monaco  ...                   0.0   60.5
4790     Diogo Costa            Porto  ...                   0.0  185.5
3926          H. Lak       Persepolis  ...                   0.0    7.0
530         G. Rulli       Villarreal  ...                   0.0   14.0
1572     P. Niazmand          Sepahan  ...                   0.0   71.5
2349      C. Maxwell        Blackpool  ...                   0.0   26.0
3148    V. Kovačević         Sarajevo  ...                   0.0   10.5
4516        O. Kolář     Slavia Praha  ...                 100.0   22.5
2602    J. Griffiths  Cheltenham Town  ...                   0.0   29.0
2689     D. Bouzanis    Sutton United  ...                   0.0   33.0
1227     Rubén Yáñez           Getafe  ...                   0.0  125.0
3311       I. Ivušić           Osijek  ...                   0.0   12.0
3541        A. Lunin      Real Madrid  ...                   0.0   45.5
4147   A. Iannarilli          Ternana  ...                   0.0   19.5

[20 rows x 114 columns]

In [46]:
#Creating DataFrame for Left Back
LB_df = dataframe[(dataframe['1st Position'] ==  "LB")]

In [47]:
LB_df.head(10)

Player           Team  ... Penalties taken Penalty conversion, %
2           A. Turan     Ankaragücü  ...               0                   0.0
33          D. Potts     Luton Town  ...               0                   0.0
48    Carlos Pomares       Tenerife  ...               0                   0.0
56   K. Hovhannisyan         Kairat  ...               0                   0.0
57         K. Haugen          Molde  ...               0                   0.0
68        I. Panţîru  FCS Bucureşti  ...               0                   0.0
78      K. Krizmanić         Gorica  ...               0                   0.0
117          V. Mora       Tlaxcala  ...               0                   0.0
134        I. Diallo        Ajaccio  ...               0                   0.0
157      D. Mosquera       Santa Fe  ...               0                   0.0

[10 rows x 113 columns]

In [58]:
#Sorting values for LeftBack based on selected features.
LB_df = LB_df.sort_values(["Defensive duels per 90", "Defensive duels won, %","Aerial duels per 90","Aerial duels won, %","Sliding tackles per 90","Accurate passes to final third, %","Deep completed crosses per 90"], ascending = (False, False,False, False,False, False,False))

In [59]:
LB_df.head(10)

Player             Team  ... Penalty conversion, %   Rank
419     N. Zolotov  Kolos Kovalivka  ...                   0.0  318.0
1178         Fábio           Nantes  ...                   0.0  346.0
2437      N. Hefti             Thun  ...                   0.0  294.5
1181     C. Traoré           Nantes  ...                   0.0  123.0
5385     J. Robson    Dundee United  ...                   0.0   59.5
5400     Y. Fukuda      Gamba Osaka  ...                   0.0  147.0
2906  T. Pellenard     Valenciennes  ...                   0.0  101.5
4580   T. Mitchell   Crystal Palace  ...                   0.0  247.0
586      F. Ortega  Vélez Sarsfield  ...                   0.0  287.0
2605      B. Conus            Aarau  ...                   0.0  275.0

[10 rows x 114 columns]

In [60]:
#Ranking Leftback based on "Defensive duels won, %"
LB_df["Rank"] = LB_df["Defensive duels won, %"].rank()
LB_df.head(20)

Player              Team  ... Penalty conversion, %   Rank
419       N. Zolotov   Kolos Kovalivka  ...                 0.000  318.0
1178           Fábio            Nantes  ...                 0.000  346.0
2437        N. Hefti              Thun  ...                 0.000  294.5
1181       C. Traoré            Nantes  ...                 0.000  123.0
5385       J. Robson     Dundee United  ...                 0.000   59.5
5400       Y. Fukuda       Gamba Osaka  ...                 0.000  147.0
2906    T. Pellenard      Valenciennes  ...                 0.000  101.5
4580     T. Mitchell    Crystal Palace  ...                 0.000  247.0
586        F. Ortega   Vélez Sarsfield  ...                 0.000  287.0
2605        B. Conus             Aarau  ...                 0.000  275.0
802        J. Angulo             Atlas  ...                 0.000  307.0
3675       B. Pittón       San Lorenzo  ...                 0.000  326.0
4626  M. Mittelstädt        Hertha BSC  ...                 0.000  116.0
559        D. Laxalt            Celtic  ...                 0.000  244.5
4123   Matheus Leoni          Várda SE  ...                 0.000  338.0
2109      E. Alioski      Leeds United  ...                66.667  149.0
4670      Javi Galán            Huesca  ...                 0.000  316.0
2002         A. Caci        Strasbourg  ...                 0.000  308.0
1025       A. Espino             Cádiz  ...                 0.000  255.0
1288        C. Bravo  Portland Timbers  ...                 0.000  252.0

[20 rows x 114 columns]

In [61]:
#Creating DataFrame for RightBack position
RB_df = dataframe[(dataframe['1st Position'] ==  "RB")]

In [62]:
#Sorting DataFrame based on selected features for RighBack
RB_df = RB_df.sort_values(["xA per 90", "Crosses per 90","Accurate crosses, %","Dribbles per 90","Successful dribbles, %","Progressive runs per 90","Accelerations per 90"], ascending = (False, False,False, False,False, False,False))

In [67]:
#Ranking DataFrame based on "Accurate crosses, %"
RB_df["Rank"] = RB_df["Accurate crosses, %"].rank()
RB_df.head(20)

Player                Team  ... Penalty conversion, %   Rank
5307     J. Tavernier             Rangers  ...                83.333  166.0
5280  Diogo Gonçalves             Benfica  ...                 0.000  286.0
2444         A. Forde       Oxford United  ...                 0.000  429.0
662         J. Murphy    Newcastle United  ...                 0.000  377.0
5657  C. Borchgrevink           Vålerenga  ...                 0.000  424.0
2194       S. Klaiber                Ajax  ...                 0.000  396.5
3809          Arnaldo         Atlético GO  ...                 0.000  347.0
2818    Lucas Vázquez         Real Madrid  ...                 0.000  314.0
2039        H. Siquet  Standard Liège U21  ...                 0.000  406.0
3093      K. Bartolec           København  ...                 0.000  202.5
3341    R. Fredericks     West Ham United  ...                 0.000  438.0
5018       R. Aguilar              Monaco  ...                 0.000  321.0
916      M. Gaćinović          Hoffenheim  ...                 0.000  168.5
1435     O. Danchenko          AEK Athens  ...                 0.000  319.5
2618     B. van Rooij                 NEC  ...                 0.000  368.0
1890       R. Çivelek         Kayserispor  ...                 0.000  336.0
360         J. Zimmer      Kaiserslautern  ...                 0.000  307.0
3020    A. Czerwiński         Lech Poznań  ...                 0.000  214.0
2113          K. Tete              Fulham  ...                 0.000  394.0
953     Igor Vinícius           São Paulo  ...                 0.000  304.0

[20 rows x 114 columns]

In [81]:
#Eliminating redundant columns
RB_df1 = RB_df[['Player', 'Team', 'Position', '1st Position', 'Age', 'Market value', "xA per 90", "Crosses per 90","Accurate crosses, %","Dribbles per 90","Successful dribbles, %","Progressive runs per 90","Accelerations per 90",'Matches played', 'Minutes played', 'Goals', 'Rank']].copy()

In [82]:
RB_df1.head(20)

Player                Team  ... Goals   Rank
5307     J. Tavernier             Rangers  ...    19  166.0
5280  Diogo Gonçalves             Benfica  ...     4  286.0
2444         A. Forde       Oxford United  ...     1  429.0
662         J. Murphy    Newcastle United  ...     7  377.0
5657  C. Borchgrevink           Vålerenga  ...     2  424.0
2194       S. Klaiber                Ajax  ...     1  396.5
3809          Arnaldo         Atlético GO  ...     1  347.0
2818    Lucas Vázquez         Real Madrid  ...     2  314.0
2039        H. Siquet  Standard Liège U21  ...     0  406.0
3093      K. Bartolec           København  ...     2  202.5
3341    R. Fredericks     West Ham United  ...     1  438.0
5018       R. Aguilar              Monaco  ...     2  321.0
916      M. Gaćinović          Hoffenheim  ...     1  168.5
1435     O. Danchenko          AEK Athens  ...     0  319.5
2618     B. van Rooij                 NEC  ...     4  368.0
1890       R. Çivelek         Kayserispor  ...     1  336.0
360         J. Zimmer      Kaiserslautern  ...     0  307.0
3020    A. Czerwiński         Lech Poznań  ...     0  214.0
2113          K. Tete              Fulham  ...     0  394.0
953     Igor Vinícius           São Paulo  ...     2  304.0

[20 rows x 17 columns]

In [83]:
#Creating DataFrame for Left Center back position
LCB_df = dataframe[(dataframe['1st Position'] ==  "LCB")]

In [85]:
LCB_df.head()

Player               Team  ... Penalties taken Penalty conversion, %
3      P. Bourdin  Beerschot-Wilrijk  ...               0                   0.0
20      P. Ouaneh      Concarneau US  ...               0                   0.0
24        Maracás  Paços de Ferreira  ...               0                   0.0
43  R. Nicolaisen         Portsmouth  ...               0                   0.0
44      K. Medina            Qarabag  ...               0                   0.0

[5 rows x 113 columns]

In [87]:
#Eliminating redundant columns
LCB_df1 = LCB_df[['Player', 'Team', 'Position', '1st Position', 'Age', 'Market value', "Successful defensive actions per 90", "Passes per 90","Accurate passes, %","Forward passes per 90","Accurate forward passes, %","Short / medium passes per 90","Accurate short / medium passes, %",'Matches played', 'Minutes played', 'Goals']].copy()

In [88]:
LCB_df1.head()

Player               Team  ... Minutes played Goals
3      P. Bourdin  Beerschot-Wilrijk  ...           2280     3
20      P. Ouaneh      Concarneau US  ...           3070     4
24        Maracás  Paços de Ferreira  ...           2876     2
43  R. Nicolaisen         Portsmouth  ...           2214     1
44      K. Medina            Qarabag  ...           3190     0

[5 rows x 16 columns]

In [89]:
#Sorting DataFrame based on selected features for Left Center Back
LCB_df1 = LCB_df1.sort_values(["Successful defensive actions per 90", "Passes per 90","Accurate passes, %","Forward passes per 90","Accurate forward passes, %","Short / medium passes per 90","Accurate short / medium passes, %"], ascending = (False, False,False, False,False, False,False))

In [92]:
#Ranking DataFrame based on "Accurate passes, %"
LCB_df1["Rank"] = LCB_df1["Accurate passes, %"].rank()
LCB_df1.head(20)

Player                 Team  ... Goals   Rank
2794         M. Senesi            Feyenoord  ...     3  187.0
1162         K. Pierie               Twente  ...     0  102.5
3390     Nurio Fortuna                 Gent  ...     2   47.0
334          N. Sibisi        Golden Arrows  ...     0  118.0
1079          K. Bryan     Sheffield United  ...     1   23.0
5493         E. Haugan        Östersunds FK  ...     1  240.0
2500       N. Havenaar                 Thun  ...     5  206.5
5609        Léo Coelho                Fénix  ...     1   19.5
3055           B. Okoh            Liefering  ...     1   29.0
1491  J. Van den Bergh    Beerschot-Wilrijk  ...     4   30.0
5667       D. Gugganig  WSG Swarovski Tirol  ...     2   60.0
1012          G. Ortíz     Atlético Tucumán  ...     2   43.0
5166            Bremer               Torino  ...     7  206.5
5690      S. Bryzgalov                Fakel  ...     4   73.5
4018            Y. Rak               Torque  ...     4  201.0
3044    D. Affengruber            Liefering  ...     4   25.0
3643         D. Hancko         Sparta Praha  ...     7   86.0
415    A. Gryszkiewicz        Górnik Zabrze  ...     0   37.0
615           J. Panzo                Dijon  ...     0  346.5
4538        B. Diakité             Toulouse  ...     2  288.0

[20 rows x 17 columns]

In [96]:
#Creating DataFrame for Mid Fielder position
MF_df = dataframe[(dataframe['1st Position'] ==  "LCMF")]

In [101]:
#Eliminating redundant columns
MF_df1 = MF_df[['Player', 'Team', 'Position', '1st Position', 'Age', 'Market value','Matches played', 'Minutes played', 'Goals', "Passes per 90","Accurate passes, %","Forward passes per 90","Accurate forward passes, %","Short / medium passes per 90","Accurate short / medium passes, %","Long passes per 90","Accurate long passes, %","Passes to final third per 90","Accurate passes to final third, %","Progressive passes per 90","Accurate progressive passes, %"]].copy()

In [102]:
#Sorting DataFrame based on selected features for Mid Fielder
MF_df1 = MF_df1.sort_values(["Passes per 90","Accurate passes, %","Forward passes per 90","Accurate forward passes, %","Short / medium passes per 90","Accurate short / medium passes, %","Long passes per 90","Accurate long passes, %","Passes to final third per 90","Accurate passes to final third, %","Progressive passes per 90","Accurate progressive passes, %"], ascending = (False))

In [103]:
#Ranking DataFrame based on "Accurate passes, %"
MF_df1["Rank"] = MF_df1["Accurate passes, %"].rank()
MF_df1.head(20)

Player                Team  ... Accurate progressive passes, %   Rank
2821    M. Verratti                 PSG  ...                          87.76  434.0
2339       H. Akino    V-Varen Nagasaki  ...                          85.08  435.0
1772   J. McEachran  Milton Keynes Dons  ...                          82.76  389.0
656      G. Cuéllar            Al Hilal  ...                          88.27  442.0
5155     B. Gilmour             Chelsea  ...                          77.33  425.0
149    E. Andersson          Djurgården  ...                          79.17  346.0
4594    C. McGregor              Celtic  ...                          85.06  428.0
1580       K. Fukai   Consadole Sapporo  ...                          78.18  438.0
1799  Yong-Woo Park     Gimcheon Sangmu  ...                          82.41  352.0
5532    R. Harakawa        Cerezo Osaka  ...                          77.48  421.5
4769      G. Kamara             Rangers  ...                          83.87  433.0
5241     I. Sangaré                 PSV  ...                          81.93  363.0
5502     B. Vrhovec             Maribor  ...                          82.59  404.0
523     André Horta      Sporting Braga  ...                          87.71  395.0
1939    Rúben Pinto            Fehérvár  ...                          83.11  345.0
4950       S. Sensi      Internazionale  ...                          88.37  356.0
4108    Y. Yamamoto         Gamba Osaka  ...                          76.05  288.0
2203           Tozé             Al Nasr  ...                          83.70  376.0
2980       A. Yusuf        IFK Göteborg  ...                          78.10  410.0
4968    E. Palacios    Bayer Leverkusen  ...                          81.13  390.0

[20 rows x 22 columns]

In [104]:
#Creating DataFrame for DMF position
DMF_df = dataframe[(dataframe['1st Position'] ==  "DMF")]

In [108]:
#Eliminating redundant columns
DMF_df1 = DMF_df[['Player', 'Team', 'Position', '1st Position', 'Age', 'Market value','Matches played', 'Minutes played', 'Goals', "Defensive duels per 90", "Defensive duels won, %","Sliding tackles per 90","Interceptions per 90","Passes per 90","Accurate passes, %","Lateral passes per 90","Accurate lateral passes, %"]].copy()

In [110]:
#Sorting DataFrame based on selected features for DMF
DMF_df1 = DMF_df1.sort_values(["Defensive duels per 90", "Defensive duels won, %","Sliding tackles per 90","Interceptions per 90","Passes per 90","Accurate passes, %","Lateral passes per 90","Accurate lateral passes, %"], ascending = (False))
DMF_df1.head()

Player  ... Accurate lateral passes, %
64      E. Camalov  ...                      93.97
820       A. Cubas  ...                      93.70
1550    A. Arregui  ...                      88.14
5497    M. Mellado  ...                      90.78
314   J. Biegański  ...                      92.89

[5 rows x 17 columns]

In [111]:
#Raking DataFrame based on "Interceptions per 90"
DMF_df1["Rank"] = DMF_df1["Interceptions per 90"].rank()
DMF_df1.head(20)

Player                Team  ... Accurate lateral passes, %   Rank
64            E. Camalov                Zira  ...                      93.97  190.0
820             A. Cubas               Nîmes  ...                      93.70   88.0
1550          A. Arregui       Independiente  ...                      88.14  162.0
5497          M. Mellado                 OFI  ...                      90.78  123.0
314         J. Biegański       Lechia Gdańsk  ...                      92.89  148.0
123        T. Pietermaat   Beerschot-Wilrijk  ...                      92.26  164.0
4175       R. Fishchenko               Veles  ...                      88.59  137.0
632             J. Ferri         Montpellier  ...                      92.75    5.0
808          M. Frendrup             Brøndby  ...                      88.00  128.0
3685           C. Rivera        Club Tijuana  ...                      94.59   27.5
2814            R. Tapia       Celta de Vigo  ...                      91.97  182.5
2784     Bruno Guimarães  Olympique Lyonnais  ...                      94.80   14.0
2740        Á. Rodríguez               Fénix  ...                      87.35  150.5
4128         J. Grønning              Viborg  ...                      92.97  175.0
1480          T. Trybull        Norwich City  ...                      96.08  200.0
4445            R. Broja    Partizani Tirana  ...                      94.00  164.0
138          M. Bernauer             Le Mans  ...                      87.93  179.0
2226  É. Álvarez Balanta         Club Brugge  ...                      94.20  148.0
4174            A. Krymi    Vllaznia Shkodër  ...                      94.16   86.5
1935        K. Stratznig      Wolfsberger AC  ...                      81.05  152.0

[20 rows x 18 columns]

In [112]:
#Creating DataFrame for LWF position
LWF_df = dataframe[(dataframe['1st Position'] ==  "LWF")]

In [113]:
#Eliminating redundant columns
LWF_df1 = LWF_df[['Player', 'Team', 'Position', '1st Position', 'Age', 'Market value','Matches played', 'Minutes played', 'Goals', "xG per 90", "xA per 90","Successful attacking actions per 90","Crosses per 90","Accurate crosses, %","Dribbles per 90","Successful dribbles, %","Offensive duels per 90","Offensive duels won, %","Corners per 90"]].copy()

In [114]:
#Sorting DataFrame based on selected features for LWF
#Raking DataFrame based on "xA per 90"

LWF_df1 = LWF_df1.sort_values(["xG per 90", "xA per 90","Successful attacking actions per 90","Crosses per 90","Accurate crosses, %","Dribbles per 90","Successful dribbles, %","Offensive duels per 90","Offensive duels won, %","Corners per 90"], ascending = (False))
LWF_df1["Rank"] = LWF_df1["xA per 90"].rank()
LWF_df1.head(20)

Player                Team  ... Corners per 90  Rank
4886    K. Toko Ekambi  Olympique Lyonnais  ...           0.05  13.0
2259          A. Katai       Crvena Zvezda  ...           3.21  42.0
1950        S. Vecchia           Rosenborg  ...           0.89  31.5
3993         A. Schalk            Servette  ...           0.07  16.0
5187          J. Hauge               Milan  ...           0.41  35.0
4919     Ricardo Horta      Sporting Braga  ...           1.29  26.5
5257        V. Mihăilă               Parma  ...           0.17  16.0
4026      V. Edvardsen           Degerfors  ...           0.49  39.5
1021         J. Koroma   Huddersfield Town  ...           0.00  13.0
119          I. Kallon             Cambuur  ...           0.25  20.0
4548    S. El Shaarawy                Roma  ...           0.00  20.0
3612       P. Almqvist              Rostov  ...           0.10   7.5
4352         E. D'Ursi                Bari  ...           0.76   7.5
1164      M. Lestienne      Standard Liège  ...           1.00  39.5
2135          F. Coman       FCS Bucureşti  ...           2.43  35.0
5141  Munir El Haddadi             Sevilla  ...           0.71  24.0
4315        A. Bifulco              Padova  ...           0.00   2.0
2836    M. Pucciarelli   Dibba Al Fujairah  ...           0.00  24.0
4035        E. Rashani      BB Erzurumspor  ...           0.95   7.5
2123        D. Lesovoy       Dinamo Moskva  ...           0.44  29.0

[20 rows x 20 columns]

In [115]:
#Creating DataFrame for RWF position
RWF_df = dataframe[(dataframe['1st Position'] ==  "RWF")]

In [116]:
#Eliminating redundant columns
RWF_df1 = RWF_df[['Player', 'Team', 'Position', '1st Position', 'Age', 'Market value','Matches played', 'Minutes played', 'Goals', "xG per 90", "xA per 90","Successful attacking actions per 90","Crosses per 90","Accurate crosses, %","Dribbles per 90","Successful dribbles, %","Offensive duels per 90","Offensive duels won, %","Corners per 90"]].copy()

In [117]:
#Sorting DataFrame based on selected features for RWF
#Raking DataFrame based on "xA per 90"
RWF_df1 = RWF_df1.sort_values(["xG per 90", "xA per 90","Successful attacking actions per 90","Crosses per 90","Accurate crosses, %","Dribbles per 90","Successful dribbles, %","Offensive duels per 90","Offensive duels won, %","Corners per 90"], ascending = (False))
RWF_df1["Rank"] = RWF_df1["xA per 90"].rank()
RWF_df1.head(20)

Player                 Team  ... Corners per 90  Rank
4207      B. Gurtsiev  Alaniya Vladikavkaz  ...           1.45  27.0
4955      T. Kadewere   Olympique Lyonnais  ...           0.04   7.5
3226         G. Iancu        Akhmat Grozny  ...           1.79  16.0
1217  P. Zinckernagel              Watford  ...           4.96  45.0
5048      S. Berghuis            Feyenoord  ...           3.36  43.0
2594        K. Tagawa                Tokyo  ...           0.00   4.5
5174           D. Man                Parma  ...           0.32  22.5
150       E. Chilufya           Djurgården  ...           0.00  33.5
94       N. Shipnoski          Saarbrücken  ...           0.00  16.0
2694    E. Volpicelli             Matelica  ...           2.14  33.5
1232        A. Gruber            LASK Linz  ...           0.29  22.5
3408          A. Amoo             Hammarby  ...           0.08   7.5
4020       C. Chiricò               Padova  ...           1.90  33.5
3657     S. Gallagher     Blackburn Rovers  ...           0.00   7.5
160         Y. Sugita               Sirius  ...           0.35  14.0
1231     T. Goiginger            LASK Linz  ...           2.12  33.5
2999     M. Whittaker         Swansea City  ...           0.19   1.0
2830      David Neres                 Ajax  ...           0.46  42.0
5031       A. Januzaj        Real Sociedad  ...           3.96  38.5
4989        B. Mbeumo            Brentford  ...           1.56  30.0

[20 rows x 20 columns]

In [118]:
#Creating DataFrame for CF position
CF_df = dataframe[(dataframe['1st Position'] ==  "CF")]

In [120]:
#Eliminating redundant columns
CF_df1 = CF_df[['Player', 'Team', 'Position', '1st Position', 'Age', 'Market value','Matches played', 'Minutes played', 'Goals',"Successful attacking actions per 90","xG per 90","Shots per 90","Shots on target, %","Goal conversion, %","Touches in box per 90","Penalties taken","Penalty conversion, %"]].copy()

In [121]:
#Sorting DataFrame based on selected features for CF
#Raking DataFrame based on "xG per 90"
CF_df1 = CF_df1.sort_values(["Successful attacking actions per 90","xG per 90","Shots per 90","Shots on target, %","Goal conversion, %","Touches in box per 90","Penalties taken","Penalty conversion, %"], ascending = (False))
CF_df1["Rank"] = CF_df1["xG per 90"].rank()


Player                 Team  ... Penalty conversion, %   Rank
2773           Neymar                  PSG  ...                91.667  934.5
1033      F. Castillo         Club Tijuana  ...                 0.000   62.5
5312         A. Ounas              Crotone  ...                 0.000   45.5
3759        O. Bukari                 Gent  ...                 0.000  372.0
3615        R. Mirzov               Khimki  ...               100.000  208.0
5198      K. Sulemana         Nordsjælland  ...                50.000  521.5
1396       N. Barrios  Central Córdoba SdE  ...                 0.000    3.5
2933        V. Burgoa           Godoy Cruz  ...                 0.000   25.5
5577        J. Zwarts            Excelsior  ...                 0.000  406.0
3250       M. Phaeton             Guingamp  ...                 0.000  435.0
2843      N. Ninković              Brescia  ...                 0.000  112.0
850        G. Pereiro             Cagliari  ...                 0.000  729.5
2701     F. Galuppini               Renate  ...                66.667  435.0
585         M. Daramy            København  ...                 0.000   62.5
5095  Gerard Deulofeu              Udinese  ...                 0.000  127.0
102        G. Beridze               Újpest  ...               100.000  577.5
5368       C. Lungoyi               Lugano  ...                 0.000  341.5
2840           E. Mor        Celta de Vigo  ...                 0.000    3.5
2863         G. Laura                Paris  ...                50.000  341.5
1118        Dany Mota                Monza  ...                 0.000  372.0

[20 rows x 18 columns]

In [122]:
CF_df1.head(40)

Player                 Team  ... Penalty conversion, %   Rank
2773           Neymar                  PSG  ...                91.667  934.5
1033      F. Castillo         Club Tijuana  ...                 0.000   62.5
5312         A. Ounas              Crotone  ...                 0.000   45.5
3759        O. Bukari                 Gent  ...                 0.000  372.0
3615        R. Mirzov               Khimki  ...               100.000  208.0
5198      K. Sulemana         Nordsjælland  ...                50.000  521.5
1396       N. Barrios  Central Córdoba SdE  ...                 0.000    3.5
2933        V. Burgoa           Godoy Cruz  ...                 0.000   25.5
5577        J. Zwarts            Excelsior  ...                 0.000  406.0
3250       M. Phaeton             Guingamp  ...                 0.000  435.0
2843      N. Ninković              Brescia  ...                 0.000  112.0
850        G. Pereiro             Cagliari  ...                 0.000  729.5
2701     F. Galuppini               Renate  ...                66.667  435.0
585         M. Daramy            København  ...                 0.000   62.5
5095  Gerard Deulofeu              Udinese  ...                 0.000  127.0
102        G. Beridze               Újpest  ...               100.000  577.5
5368       C. Lungoyi               Lugano  ...                 0.000  341.5
2840           E. Mor        Celta de Vigo  ...                 0.000    3.5
2863         G. Laura                Paris  ...                50.000  341.5
1118        Dany Mota                Monza  ...                 0.000  372.0
4979         S. Koita             Salzburg  ...               100.000  953.0
2563        A. Zaizen     Wacker Innsbruck  ...                 0.000  653.5
3286      Đ. Ivanović             Olimpija  ...               100.000  286.0
1721      H. Duljević                Nîmes  ...                 0.000   13.0
2053        Á. Romero          San Lorenzo  ...                60.000  435.0
3781         Adailton                Tokyo  ...                 0.000  653.5
1668    Seon-Min Moon      Gimcheon Sangmu  ...                50.000  653.5
771         G. Rutter           Hoffenheim  ...               100.000  633.0
970           M. Elia           Young Boys  ...                 0.000  372.0
5417         J. Okita                  NEC  ...                 0.000  488.5
2768          W. Zaha       Crystal Palace  ...               100.000  170.5
1521      D. Vorobyev             Orenburg  ...               100.000  814.0
2858          J. Ngoy                Eupen  ...                 0.000  488.5
1308       T. Ambrose                 Metz  ...               100.000   62.5
4068        I. Jebali                   OB  ...               100.000  607.0
77         M. Ignatov       Spartak Moskva  ...                 0.000  372.0
825        Marquinhos        Botev Plovdiv  ...                 0.000   82.5
4478         M. Awaed        Maccabi Haifa  ...                 0.000  673.0
1287    Omar Marmoush            St. Pauli  ...               100.000  208.0
5104       K. Adeyemi             Salzburg  ...                 0.000  488.5

[40 rows x 18 columns]

We decided that we wanted to build a team of 442 with attacking wingbacks and ball playing defenders. 
Our goal keeper is R.Majecki from Monaco worth 10 million
Our left back is M.Mittelstadt from Hertha worth 7 million
Our right back is H.Siquet from standard liege worth 5 million
Our centre backs are J.Panzo from Dijon worth 4 million 
And B.Diakite from Toulouse worth 7 million

One of our centre mids is a deep lying play maker, we chose G.Cuellar from Al Hilal worth 4 million for this
The other centre mid is a ball winning mid. We chose E Alvarez Balanta from club brugge worth 4.5 million
Our left winger is there for assist and goals as well as crossing. We chose J.Hauge from Milan worth 9.5 million

Our right winger is there for a similar reason to the left winger but with more of a focus at passing. We chose D.Man from Parma worth 10 million

Our two strikers are K.Sulemana from Nordsjaelland worth 9 million and S.Koita from Salzburg worth 12.5. These players have great goals per game and xG and Koita being young has good upside and potential. 


Total cost of this team is 76.5 million
